Lib/glyphsLib/parser.py

```diff
@@ -114,17 +114,113 @@ def _parse_list(self, text, i):
         i += len(parsed)
         return res, i
 
+    # glyphs only supports octal escapes between \000 and \077
+    _unescape_re = re.compile(ur'(\\0[0-7]{2})|(\\U[0-9a-fA-F]{4,6})')
+
+    @staticmethod
+    def _unescape_fn(m):
+        if m.group(1):
+            return unichr(int(m.group(1)[1:], 8))
+        return unichr(int(m.group(2)[2:], 16))
+
     def _trim_value(self, value):
         """Trim double quotes off the ends of a value, un-escaping inner
         double quotes.
+        Also convert escapes to unicode.
         """
 
         if value[0] == '"':
             assert value[-1] == '"'
-            return value[1:-1].replace('\\"', '"')
-        return value
+            value = value[1:-1].replace('\\"', '"')
+        return Parser._unescape_re.sub(Parser._unescape_fn, value)
 
     def _fail(self, message, text, i):
         """Raise an exception with given message and text at i."""
 
         raise ValueError('%s:\n%s' % (message, text[i:i + 79]))
+
+
+class Writer(object):
+    """Write parsed data back to flat file.  Normalizes quoting
+    and indentation."""
+
+    # ints and floats are unquoted, as are strings of letters/digits and
+    # underscore with period or leading forward slash.  Everything else
+    # is quoted.
+    _sym_re = re.compile(
+        r'^(?:-?[0-9]+\.?[0-9]*|[_a-zA-Z0-9/\.][_a-zA-Z0-9\.]*)$')
+
+    def __init__(self, out=sys.stdout, indent=0, reorder=False):
+        self.out = out
+        self.indent = indent
+        self.reorder = reorder
+        self.curindent = 0
+
+    def write(self, data):
+        self.curindent = 0
+        self._write(data)
+        self.out.write('\n')
+        self.out.flush()
+
+    def _write(self, data):
+        if isinstance(data, dict):
+            self._write_dict(data)
+        elif isinstance(data, list):
+            self._write_list(data)
+        else:
+            self._write_atom(data)
+
+    def _write_dict(self, data):
+        if self.reorder:
+            keys = sorted(data.keys())
+        else:
+            keys = data.keys()
+        self.curindent += self.indent
+        pad = ' ' * self.curindent
+        out = self.out
+        out.write('{\n')
```

Comment:
We need `out.write(' ' * self.curindent)` before this line too if I'm not mistaken (and to increment curindent after this line instead of before).